tools/tu_collector/tu_collector/tu_collector.py

```diff
@@ -51,6 +52,40 @@ def __get_toolchain_compiler(command):
                                 'g++' if is_cpp else 'gcc')
 
 
+def __determine_compiler(gcc_command):
```

Comment:
If you add `find_executable` function as a function argument to the `__determine_compiler` you can easily unit test this function.
In the test the `find_executable_mock` would just return some value or None depending on the test.